src/aleph/vm/pool.py

```diff
@@ -340,3 +350,66 @@ def get_executions_by_sender(self, payment_type: PaymentType) -> dict[str, dict[
                 executions_by_sender.setdefault(sender, {})
                 executions_by_sender[sender].setdefault(chain, []).append(execution)
         return executions_by_sender
+
+    class Reservation:
```

Comment:
I think that we can move that class outside the `VMPool` class as it's more generic